shell/browser/javascript_environment.cc

```diff
@@ -73,21 +73,33 @@ struct base::trace_event::TraceValue::Helper<
 
 namespace electron {
 
-JavascriptEnvironment::JavascriptEnvironment(uv_loop_t* event_loop)
+namespace {
+auto CreateIsolateHolder(v8::Isolate* isolate, NodeEnvironmentType type) {
+  std::unique_ptr<v8::Isolate::CreateParams> create_params =
+      gin::IsolateHolder::getDefaultIsolateParams();
+
+  if (type == NodeEnvironmentType::kNodeMode) {
+    create_params->only_terminate_in_safe_scope = false;
```

Comment:
Might be worth explaining why this works in the PR description?

Reply:
done